google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDMLTest.java

```diff
@@ -83,6 +87,11 @@ public Long run(TransactionContext transaction) {
             long rowCount = 0;
             for (String stmt : stmts) {
               rowCount += transaction.executeUpdate(Statement.of(stmt));
+
```

Comment:
This block should probably be moved up before transaction.executeUpdate(..). Otherwise, the DML will be executed twice in the backend, second time being a replay.

Reply:
Fixed.

Yes, I agree about the unit test. There is no DML unit test and it's going to be a much bigger change than we want right now for GA.

Reply:
Hm, actually that doesn't work. Looks like there are some assumptions that are affected.

com.google.cloud.spanner.SpannerException: UNKNOWN: Waited 500 milliseconds for com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture@73a2e526[status=PENDING, info=[inputFuture=[com.google.api.core.ApiFutureToListenableFuture@7d64e326], exceptionType=[class java.lang.Exception], fallback=[com.google.api.core.ApiFutures$GaxFunctionToGuavaFunction@13f95696]]]

Reply:
Nm, sorry for the spam. It was unrelated.